erpnext/projects/doctype/timesheet/timesheet.py

```diff
@@ -512,7 +512,7 @@ def get_timesheets_list(doctype, txt, filters, limit_start, limit_page_length=20
 		sales_invoices = [
 			d.name for d in frappe.get_all("Sales Invoice", filters={"customer": customer})
 		] or [None]
-		projects = [d.name for d in frappe.get_all("Project", filters={"customer": customer})]
+		projects = [d.name for d in frappe.get_all("Project", filters={"customer": customer})] or [None]
```

Comment:
if no projects then don't add project condition in the sql query. 

Better write this 
```
projects = frappe.get_all("Project", filters={"customer": customer}, pluck="name")
```